xml/mysql_banners.xml

```diff
@@ -0,0 +1,812 @@
+<?xml version="1.0"?>
+<fingerprints>
```

Comment:
You should add a `matches` attribute here to indicate what it should be matching.  

That brings up something I think we discussed in email but I don't see addressed here.  This single file matches against two different types of data -- version banners and error messages.  IMO these should go in separate files and use different `match` attributes.  Then, anyone/anything using these files can use the right file depending on the type of message received.